### PR TITLE
msg: always generate random nonce; don't try to reuse PID

### PR DESCRIPTION
**Pull request by rzarzynski**
We'll need to drop the env. var. from Rook's code base as well.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by adk3798 — APPROVED** [review]
I don't know the story behind why we're doing this, but the cephadm change does look to correctly be removing the env variable as you're intending. So can approve that bit at least.

File: src/msg/Messenger.cc [hunk 1]
```diff
@@ -20,12 +20,12 @@ Messenger *Messenger::create_client_messenger(CephContext *cct, std::string lnam
...
-    // we're running in a container; use a random number instead!
-    nonce = ceph::util::generate_random_number<uint64_t>();
-  }
-  return nonce;
+  // in the past the logic here was more complex -- we were trying
+  // to use the PID but, in the containerized world, it turned out
+  // unreliable. To deal with this, we started guessing whether we
+  // run in a container or not, and of course, got manual lever to
+  // intervene if guessed wrong (CEPH_USE_RANDOM_NONCE).
+  return ceph::util::generate_random_number<uint64_t>();
```
Comment (idryomov) [inline src/msg/Messenger.cc:28]: Why not just nuke `get_pid_nonce`, replacing all calls with `get_random_nonce`?

Comment (rzarzynski) [reply src/msg/Messenger.cc:28]: :+1:.

File: src/crimson/osd/main_config_bootstrap_helpers.h [hunk 2]
```diff
@@ -21,14 +21,9 @@ void usage(const char* prog);
...
-    // we're running in a container; use a random number instead!
-    std::random_device rd;
-    std::default_random_engine rng{rd()};
-    return std::uniform_int_distribution<uint64_t>{}(rng);
-  } else {
-    return pid;
-  }
+  std::random_device rd;
+  std::default_random_engine rng{rd()};
+  return std::uniform_int_distribution<uint64_t>{}(rng);
```
Comment (idryomov) [inline src/crimson/osd/main_config_bootstrap_helpers.h:26]: Can this be `ceph::util::generate_random_number<uint64_t>()` for consistency?  Or even call `Messenger::get_random_nonce()` directly from crimson?

**Review by idryomov — APPROVED** [review]

**Comment by idryomov** [comment]
jenkins test windows

**Comment by idryomov** [comment]
jenkins test dashboard cephadm

**Comment by ljflores** [comment]
Rados suite review: https://pulpito.ceph.com/?branch=wip-yuri3-testing-2023-04-25-1147

Failures, unrelated:
    1. https://tracker.ceph.com/issues/59049
    2. https://tracker.ceph.com/issues/59192
    3. https://tracker.ceph.com/issues/59335
    4. https://tracker.ceph.com/issues/59380
    5. https://tracker.ceph.com/issues/58946
    6. https://tracker.ceph.com/issues/50371
    7. https://tracker.ceph.com/issues/59057
    8. https://tracker.ceph.com/issues/53345

Details:
    1. WaitReplicas::react(const DigestUpdate&): Unexpected DigestUpdate event - Ceph - RADOS
    2. cls/test_cls_sdk.sh: Health check failed: 1 pool(s) do not have an application enabled (POOL_APP_NOT_ENABLED)
    3. Found coredumps on smithi related to sqlite3Found coredumps on smithi related to sqlite3 - Ceph - Cephsqlite
    4. rados/singleton-nomsgr: test failing from "Health check failed: 1 full osd(s) (OSD_FULL)" and "Health check failed: 1 filesystem is offline (MDS_ALL_DOWN)" - Ceph - RADOS
    5. cephadm: KeyError: 'osdspec_affinity' - Ceph - Mgr - Dashboard
    6. Segmentation fault (core dumped) ceph_test_rados_api_watch_notify_pp - Ceph - RADOS
    7. rados/test_envlibrados_for_rocksdb.sh: No rule to make target 'rocksdb_env_librados_test' on centos 8 - Ceph - RADOS
    8. Test failure: test_daemon_restart (tasks.cephadm_cases.test_cli.TestCephadmCLI) - Ceph - Orchestrator

**Pull request merged**